lib/segment/src/index/field_index/index_selector.rs

```diff
@@ -49,6 +49,11 @@ pub fn index_selector(
                 field,
             ))],
             PayloadSchemaType::Bool => vec![FieldIndex::BinaryIndex(BinaryIndex::new(db, field))],
+            PayloadSchemaType::Datetime => {
+                vec![FieldIndex::DatetimeIndex(
+                    NumericIndex::<IntPayloadType>::new(db, field, is_appendable),
+                )]
```

Comment:
What about another index here based on `FieldIndex::IntMapIndex(MapIndex::new(...))`?

I think we need it to properly support direct lookups. Or do we not support direct lookups on dates anyway?

Reply:
Only the range query for dates is supported. I think, `MapIndex` shouldn't be used here?

Reply:
> Only the range query for dates is supported

Then yes, thank you for confirming!